Improve Airtable plugin design

madebyisaacr opened the pull request:
### Description

<!-- What is this PR doing? Please write a clear description or reference the issues it solves (e.g. `fixes #123`). Are there any parts you think require specific attention from reviewers? -->

This pull request improves the design and UX of the Airtable plugin.

- Increase default size of plugin window to prevent text getting cut off.
- Improve appearance and sorting of unsupported fields.
- Make field types with one option non-clickable.
- Add heading that links to the Airtable base.
- Use primary button style.

Before and after:
![image](https://github.com/user-attachments/assets/0a31b1c2-93e7-4bf7-9f8c-9e2bc4849915)

melissarofman reviewed (changes requested):
Left some feedback on the code itself, but also have some questions about the redesign:
1. I'm not conviced about removing one of the carets in the field rows
2. I don't think it's necessary to say "Airtable Column"

madebyisaacr commented on plugins/airtable/src/FieldMapping.tsx:
```diff
@@ -24,14 +25,14 @@ function ChevronIcon() {
-    { type: "number", label: "Number" },
-    { type: "string", label: "String" },
+    { type: "string", label: "Plain Text" },
     { type: "formattedText", label: "Formatted Text" },
-    { type: "image", label: "Image" },
-    { type: "link", label: "Link" },
     { type: "date", label: "Date" },
+    { type: "link", label: "Link" },
+    { type: "image", label: "Image" },
+    { type: "color", label: "Color" },
+    { type: "boolean", label: "Toggle" },
+    { type: "number", label: "Number" },
     { type: "enum", label: "Option" },
     { type: "file", label: "File" },
 ]
```
This change makes the field type names and order match the field editing menu in Framer.

Boolean -> Toggle
String -> Plain Text

<img width="207" alt="image" src="https://github.com/user-attachments/assets/79a27d5a-916a-4138-903b-0ed9e4af92cd" />

madebyisaacr commented on plugins/airtable/src/FieldMapping.tsx:
```diff
@@ -41,26 +42,53 @@ interface FieldMappingRowProps {
         isIgnored,
         disabled,
+        unsupported,
+        missingCollection,
```
`missingCollection` is used for Airtable reference fields that don't have a corresponding collection in Framer.

Line 310-312 in FieldMapping.tsx
```tsx
const missingCollectionFields = fields.filter(
    field => isCollectionReference(field) && field.supportedCollections.length === 0
)
```

madebyisaacr commented on plugins/airtable/src/FieldMapping.tsx:
```diff
@@ -41,26 +42,53 @@ interface FieldMappingRowProps {
                 <button
                     type="button"
-                    className="source-field"
+                    className={cx("source-field", unsupported || missingCollection ? "unsupported" : "")}
```
Ok, removed.

madebyisaacr commented:
> I'm not conviced about removing one of the carets in the field rows

The type and name columns are combined without a caret because the left side is Airtable and right side is Framer. The Notion and Sheets plugins don't have a second caret between type and name either.

> I don't think it's necessary to say "Airtable Column"

Good point, I changed it back to just "Column"

The before and after screenshot is up to date now.

madebyisaacr commented:
Done. Fixed the linter type error, rebased, and ran the biome command:

`yarn biome format plugins/airtable --write`
`Formatted 15 files in 17ms. No fixes applied.`

madebyisaacr commented on plugins/airtable/src/FieldMapping.tsx:
```diff
@@ -325,14 +352,10 @@ export function FieldMapping({ collection, dataSource, initialSlugFieldId }: Fie
-                            field.type !== "unsupported" &&
-                            (!isCollectionReference(field) || field.supportedCollections.length > 0)
-                    )
+                    .filter(field => !unsupportedFields.includes(field) && !missingCollectionFields.includes(field))
```
`unsupportedFields` and `missingCollectionFields` are filtered arrays of `fields`, so it wouldn't be more or less reliable than the current method.

```
const unsupportedFields: PossibleField[] = fields.filter(field => field.type === "unsupported")
const missingCollectionFields: PossibleField[] = fields.filter(
    field => isCollectionReference(field) && field.supportedCollections.length === 0
)
```

triozer replied:
It's not entirely accurate to say there's no difference. Using object reference equality (i.e., `unsupportedFields.includes(field)`) will only work if the `field` objects in all arrays are the exact same instance (reference) in memory. If, for any reason, new objects are created or fields are fetched/constructed separately, then checking by reference can fail—even if the objects have the same `id` and properties.

Comparing by `field.id` (or another unique identifier) is generally more robust and reliable, especially if there's any chance that the arrays might contain different object instances representing the same logical field.

madebyisaacr replied:
Ok, I changed it.

triozer commented on plugins/airtable/src/FieldMapping.tsx:
```diff
@@ -63,61 +90,46 @@ const FieldMappingRow = memo(
-                            {field.supportedCollections.map(collection => (
-                                <option key={collection.id} value={collection.id}>
-                                    {collection.name}
+                {unsupported || missingCollection ? (
```
OK! Also, we don't use Prettier anymore, so it seems to be a Biome thing 👌🏽

triozer commented on plugins/airtable/src/App.tsx:
```diff
@@ -23,11 +23,11 @@ export function App({ collection, previousBaseId, previousTableId, previousSlugF
 
     useLayoutEffect(() => {
-        const hasDataSourceSelected = Boolean(dataSource)
+        const hasDataSourceSelected = Boolean(dataSource) || isLoadingDataSource
```
Feels like we should keep the window fixed until we have something to display, otherwise the resizing just adds noise.

madebyisaacr replied:
Yes, preventing resizing is what this change does. Previously it would resize when it finished loading.
